### ISSUE TITLE
Consolidate Rust and AssemblyScript examples

### ISSUE THREAD

**Issue by vgrichina** [nearprotocol/create-near-app#86]
Currently Rust and AssemblyScript examples use slightly different contracts and `package.json` files. This means that it's hard to make various flags orthogonal (e.g. have Rust example with React) and multiplies the number of templates we need to maintain to support such combinations.

I propose following changes:
- make contracts use the same API (and so web parts completely compatible)
- move differences in build scripts out of `package.json`

Then we can easily re-combine different backend contracts (Rust and AS) with different frontends (vanilla, React, etc) and so reduce number of templates that need to be maintained.

**Comment by vgrichina** [comment]
Also right now for AS example we only have integration tests in JS, but for Rust example – only unit tests in Rust.
Makes sense to have both type of tests for both languages. My understanding is that @willemneal is already working on as-pect support. So we should use that for AS unit tests.
We can use same JS integration tests for both languages once we have fully matching contract API.

**Comment by janedegtiareva** [comment]
Asc examples only have a view function. We should add a change function as well as part of this effort.

**Comment by janedegtiareva** [comment]
We also need to add an example of returning a user defined struct to rust.

**Comment by janedegtiareva** [comment]
https://github.com/nearprotocol/create-near-app/pull/94

**Comment by janedegtiareva** [comment]
We need to generate readme, package, and a few other things from template. Current WIP proposal is here https://github.com/nearprotocol/create-near-app/pull/100

**Comment by janedegtiareva** [comment]
We decided to punt on readme generation for now. Can we close this issue @vgrichina ?

**Comment by vgrichina** [comment]
@janedegtiareva do I understand correctly that this part is done:
> make contracts use the same API (and so web parts completely compatible)

but this one isn't:

> move differences in build scripts out of package.json

?

I think @chadoh was thinking regarding using GitHub repos as templates for `create-near-app` as well. If it's the direction we are going – makes sense to close this and replace by another tracking issue to use GH repos as templates.

**Comment by janedegtiareva** [comment]
@vgrichina  Yes, that's the correct understanding of the current state. The web parts are compatible, the only diffs are in the build scripts (+ readme files)
